### ISSUE TITLE
Aot run crash on Android

### ISSUE THREAD

**Issue by Coloryr** [AvaloniaUI/Avalonia#13250]
## Describe the bug
Run AOT build apk crash on Android.

## To Reproduce

1. Build apk with Aot
2. install apk
3. Run

## Expected behavior
logcat
```
2023-10-14 20:37:33.822 32421-32421 ndroid:launcher         coloryr.colormc.android              E  * Assertion at /__w/1/s/src/mono/mono/mini/aot-runtime.c:3699, condition `is_ok (error)' not
                                                                                                    met, function:decode_patch, module 'ColorMC.Gui.dll.so' is unusable (GUID of dependent assembly Avalonia.Controls.DataGrid
                                                                                                    doesn't match (expected '2545F13E-88B9-4A54-B15A-EAE95E4DD675', got '55F408B8-5FE2-4053-80CA-A9AFBC17EA24')).
2023-10-14 20:37:33.822 32421-32421 libc                    coloryr.colormc.android              A  Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 32421 (ndroid:launcher), pid 32421 (ndroid:launcher)
```

## Screenshots
none.

## Environment 

- OS: Android
- Avalonia-Version: 11.0.5-rc1

## Additional context
```
<ItemGroup>
  <PackageReference Include="Avalonia" Version="11.0.5-rc1" />
  <PackageReference Include="Avalonia.AvaloniaEdit" Version="11.0.1" />
  <PackageReference Include="Avalonia.Controls.ColorPicker" Version="11.0.5-rc1" />
  <PackageReference Include="Avalonia.Controls.DataGrid" Version="11.0.5-rc1" />
  <PackageReference Include="Avalonia.Controls.TreeDataGrid" Version="11.0.1" />
```

```
 <ItemGroup>
   <PackageReference Include="Avalonia.Android" Version="11.0.5-rc1" />
   <PackageReference Include="Xamarin.AndroidX.AppCompat" Version="1.6.1.4" />
   <PackageReference Include="Xamarin.AndroidX.Core" Version="1.12.0.1" />
<PackageReference Include="Xamarin.AndroidX.Core.SplashScreen" Version="1.0.1.3" />
<PackageReference Include="Xamarin.AndroidX.Preference" Version="1.2.1.1" />
<PackageReference Include="Xamarin.Google.Android.Material" Version="1.9.0.3" />
 </ItemGroup>	
```

**Comment by Coloryr** [comment]
11.0.4 can run, 11.0.5-rc1 crash

**Comment by Coloryr** [comment]
fix in 11.0.5

**Comment by Amenti** [comment]
I'm seeing this error again with the nightly build 11.2.999-cibuild0051578-alpha (I'd guess this happens on 11.1.3 as well).
